BUILDING.md

```diff
@@ -77,48 +75,48 @@ $ cd grpc/third_party/protobuf
 $ sudo make install   # 'make' should have been run by core grpc
 ```
 
-# Build from Source
+# Clone the repository (including submodules)
 
-For developers who are interested to contribute, the following commands show how to compile the
-gRPC C Core library.
+Before building, you need to clone the gRPC github repository and download submodules containing source code 
+for gRPC's dependencies (that's done by the `submodule` command or `--recursive` flag). The following commands will clone the gRPC
+repository at the latest stable version.
+
+## Unix
 
 ```sh
  $ git clone -b $(curl -L https://grpc.io/release) https://github.com/grpc/grpc
  $ cd grpc
  $ git submodule update --init
- $ make
- $ [sudo] make install
-```
+ ```
 
 ## Windows
 
-There are several ways to build under Windows, of varying complexity depending
-on experience with the tools involved.
+```
+> @rem You can also do just "git clone --recursive -b THE_BRANCH_YOU_WANT https://github.com/grpc/grpc"
+> powershell git clone --recursive -b ((New-Object System.Net.WebClient).DownloadString(\"https://grpc.io/release\").Trim()) https://github.com/grpc/grpc
+> cd grpc
+> @rem To update submodules at later time, run "git submodule update --init"
+```
 
+# Build from source
 
+## make (on UNIX systems)
 
-### Building using CMake (RECOMMENDED)
+From the grpc repository root
+```sh
+ $ make
+ # WARNING: make doesn't provide an easy way to uninstall
```

Comment:
Follow up q; I thought we do support `make uninstall`, is that not a sufficient uninstall? Might be work clarifying, because this comment, as is, is a little scary

Reply:
`make uninstall` doesn't exist (and to be honest it would be pretty hard to implement, the right solution for uninstalling is to rely on a build system / package manager that has proper support for installing/uninstalling  dependencies - I'm hoping we will get there soon).

I agree that more careful wording (and more detailed explanation) would be necessary for the warning. For now, I removed the comment and also the `make install` line as this document is about how to build (more info on installation is in src/cpp/README.md)